### PR TITLE
Merge API recommendations and short code guidance into the specification

### LINKED ISSUE #490: Please move all wiki articles into the repository // MOVE SHORT CODE SPEC
Please move all the wiki articles into the repository.

There are currently normative statements in the Wiki. Those should be under code control and should be open to discussion/PR inside this repository.

---

Requesting that these be copied as-is for now (even with broken links). And then everybody can help to clean them up together.

I myself have many corrections to make to those as per https://github.com/google/open-location-code/pull/463. And maybe if all the files are here this can help us make progress on all these things.

### PR DESCRIPTION
**Pull request by drinckes**
Fixes #490 and #432 .

This moves the API recommendations and short code guidance into the specifications document.

(Also a couple minor edits.)

### CODE REVIEW

**Review by tigger6 — APPROVED** [review]

**Review by fulldecent — APPROVED** [review]
Good to combine, a move in the right direction

**Comment by drinckes** [comment]
Thanks all for the feedback!

File: Documentation/Specification/specification.md [hunk 1]
```diff
@@ -126,19 +126,87 @@ For example, 8FVC9G8F+6W has the center 47.365562,8.524813. The following table
...
+The values above are slightly smaller than the maximums to allow for different geocoder backends placing localities in slightly different positions.
+Although they could be increased there will be a risk that a shortened code will recover to a different location than the original, and people misdirected.
+
+Note: Usually your feature will be a town or city, but you could also use geographical features such as lakes or mountains, if they are the best local reference.
+If a settlement (such as neighbourhood, town or city) is to be used, you should choose the most prominent feature that meets the requirements, to avoid using obscure features that may not be widely known.
+(Basically, prefer city or town over neighbourhood.)
+
+## API Requirements
+
+The following public methods should be provided by any Open Location Code implementation, subject to minor changes caused by language conventions.
```
Comment (fulldecent) [inline Documentation/Specification/specification.md:157]: This section is labeled "requirements" but then contradicts that using "should" below.

Please consider RFC 2119 language including "MUST".

Comment (drinckes) [reply Documentation/Specification/specification.md:157]: PR is already merged.

**Pull request merged**